lib/_tls_legacy.js

```diff
@@ -585,14 +585,12 @@ function onhandshakestart() {
   ssl.lastHandshakeTime = now;
   if (first) return;
 
-  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT) {
+  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT && self.cleartext) {
```

Comment:
Are you sure this can be safely changed (e.g. `self.cleartext`'s value won't change between here and the next tick)?